src/mono/wasm/wasm.proj

```diff
@@ -110,4 +108,109 @@
+      <EmccFlags Condition="'$(WasmEnableES6)' == 'true'">$(EmccFlags) -s MODULARIZE=1 -s EXPORT_ES6=1</EmccFlags>
+      <EmccConfigurationFlags Condition="'$(Configuration)' == 'Debug'">-g -Os -s ASSERTIONS=1 -DENABLE_NETCORE=1 -DDEBUG=1</EmccConfigurationFlags>
+      <EmccConfigurationFlags Condition="'$(Configuration)' == 'Release'">-Oz --llvm-opts 2 -DENABLE_NETCORE=1</EmccConfigurationFlags>
+      <StripCmd>$(EMSDK_PATH)/upstream/bin/wasm-opt --strip-dwarf $(NativeBinDir)/dotnet.wasm -o $(NativeBinDir)/dotnet.wasm</StripCmd>
+      <WasmObjDir>$(ArtifactsObjDir)wasm</WasmObjDir>
+      <WasmVersionFile>$(WasmObjDir)\emcc-version.txt</WasmVersionFile>
+      <MonoIncludeDir>$(MonoArtifactsPath)include/mono-2.0</MonoIncludeDir>
+      <SystemNativeDir>$(RepoRoot)src\libraries\Native\Unix\System.Native</SystemNativeDir>
+    </PropertyGroup>
+    <ItemGroup>
+      <MonoLibFiles Include="$(MonoArtifactsPath)libmono-ee-interp.a;
+                             $(MonoArtifactsPath)libmonosgen-2.0.a;
+                             $(MonoArtifactsPath)libmono-ilgen.a;
+                             $(MonoArtifactsPath)libmono-icall-table.a;
+                             $(NativeBinDir)libSystem.Native.a;
+                             $(NativeBinDir)libSystem.IO.Compression.Native.a;
+                             $(ICULibDir)/libicuuc.a;
+                             $(ICULibDir)/libicui18n.a" />
+      <PInvokeTableFile Include="$(WasmObjDir)\pinvoke-table.h" />
+      <ICULibFiles Include="$(ICULibDir)/*.dat" />
+    </ItemGroup>
+
+    <Exec Command="$(EmccCmd) $(EmccFlags) $(EmccConfigurationFlags) -Oz -I$(MonoIncludeDir) runtime/corebindings.c -c -o $(MonoObjDir)corebindings.o"
+          IgnoreStandardErrorWarningFormat="true" />
+
+    <Exec Command="$(EmccCmd) $(EmccFlags) $(EmccConfigurationFlags) -Oz -DCORE_BINDINGS -I$(MonoObjDir) -I$(MonoIncludeDir) runtime/driver.c -c -o $(MonoObjDir)driver.o"
+          IgnoreStandardErrorWarningFormat="true" />
+
+    <Copy SourceFiles="@(PInvokeTableFile)"
+          DestinationFolder="$(MonoObjDir)"
+          SkipUnchangedFiles="true" />
+
+    <Exec Command="$(EmccCmd) $(EmccFlags) $(EmccConfigurationFlags) -Oz -DGEN_PINVOKE=1 -I$(MonoObjDir) -I$(MonoIncludeDir) runtime/pinvoke.c -c -o $(MonoObjDir)pinvoke.o"
+          IgnoreStandardErrorWarningFormat="true" />
+
+    <Exec Command="$(EmccCmd) $(EmccFlags) $(EmccConfigurationFlags) --js-library runtime/library_mono.js --js-library runtime/binding_support.js --js-library runtime/dotnet_support.js --js-library $(SystemNativeDir)\pal_random.js $(MonoObjDir)/driver.o $(MonoObjDir)/pinvoke.o $(MonoObjDir)/corebindings.o @(MonoLibFiles->'%(FullPath)', ' ') -o $(NativeBinDir)/dotnet.js &amp;&amp; $(StripCmd)"
+          IgnoreStandardErrorWarningFormat="true" />
+
+    <Exec Command="$(EmccCmd) --version > $(WasmVersionFile)"
+          IgnoreStandardErrorWarningFormat="true" />
+    
+    <GenerateWasmVersionFile SourceFile="$(WasmVersionFile)"
```

Comment:
This can be done like in `WasmApp.targets`:

```xml
    <RunWithEmSdkEnv Command="emcc --version" WorkingDirectory="$(_WasmIntermediateOutputPath)" EmSdkPath="$(EMSDK_PATH)" ConsoleToMsBuild="true">
      <Output TaskParameter="ConsoleOutput" ItemName="_VersionLines" />
    </RunWithEmSdkEnv>
```